internal/v2/fakes2av2/fakes2av2_test.go

```diff
@@ -0,0 +1,177 @@
+				ReqOneof: &s2av2pb.SessionReq_GetTlsConfigurationReq {
+					&s2av2pb.GetTlsConfigurationReq {
+						ConnectionSide: commonpb.ConnectionSide_CONNECTION_SIDE_SERVER,
+					},
+				},
+			},
+			expectedResponse: &s2av2pb.SessionResp {
+				Status: &s2av2pb.Status {
+					Code: 0,
+				},
+				RespOneof: &s2av2pb.SessionResp_GetTlsConfigurationResp {
+					GetTlsConfigurationResp: &s2av2pb.GetTlsConfigurationResp {
+						TlsConfiguration: &s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration_{
+							&s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration{
+								CertificateChain: []string{
+									string(serverCert),
+								},
+								MinTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_3,
+								MaxTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_3,
+								HandshakeCiphersuites: []commonpb.HandshakeCiphersuite{},
+								RecordCiphersuites: []commonpb.RecordCiphersuite {
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_AES_128_GCM_SHA256,
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_AES_256_GCM_SHA384,
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_CHACHA20_POLY1305_SHA256,
+								},
+								TlsResumptionEnabled: false,
+								RequestClientCertificate: s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration_REQUEST_AND_VERIFY,
+								MaxOverheadOfTicketAead: 0,
+							},
+						},
+					},
+				},
+			},
+		},
+	}{
+		t.Run(tc.description, func(t *testing.T) {
+			// Start up Fake S2Av2 Server.
+			var wg sync.WaitGroup
+			wg.Add(1)
+			address, stop, err := startFakeS2Av2Server(&wg)
+			wg.Wait()
+			// Create stream to server.
+			opts := []grpc.DialOption{
+				grpc.WithTransportCredentials(insecure.NewCredentials()),
+				grpc.WithReturnConnectionError(),
+				grpc.WithBlock(),
+			}
+			conn, err := grpc.Dial(address, opts...)
+			if err != nil {
+				t.Fatalf("Client: failed to connect: %v", err)
+			}
+			defer conn.Close()
+			c := s2av2pb.NewS2AServiceClient(conn)
+			log.Printf("Client: connected to: %s", address)
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+			defer cancel()
+			// Setup bidrectional streaming session.
+			s2av2_opts := []grpc.CallOption{}
+			cstream, err0 := c.SetUpSession(ctx, s2av2_opts...)
+			if err0 != nil  {
+				t.Fatalf("Client: failed to setup bidirectional streaming RPC session: %v", err0)
+			}
+			log.Printf("Client: set up bidirectional streaming RPC session.")
+			// Send request.
+			if err := cstream.Send(tc.request); err != nil {
+				t.Fatalf("Client: failed to send SessionReq: %v", err)
+			}
+			log.Printf("Client: sent SessionReq")
+			// Get the Response.
+			resp, err1 := cstream.Recv()
+			if err1 != nil {
+				t.Fatalf("Client: failed to receive SessionResp: %v", err1)
+			}
+			log.Printf("Client: recieved SessionResp")
+			if diff := cmp.Diff(resp, tc.expectedResponse, protocmp.Transform()); diff != "" {
+				t.Errorf("cstream.Recv() returned incorrect SessionResp, (-want +got):\n%s", diff)
+			} else {
```

Comment:
I think it's OK to skip the else block - this print statement doesn't tell us anything because we already know if the test failed or not by the time we get here.